### PR TITLE
WIP: Retry settings

### PR DESCRIPTION
**Pull request by michaelbausor**
WIP PR to update retry settings. Toolkit PR: https://github.com/googleapis/toolkit/pull/1552

See comments on RetrySettings class for example usage.

cc @vam-google @dwsupplee @jdpedrie @bshaffer

### CODE REVIEW

**Comment by bshaffer** [comment]
Is there anything important to mention in this refactoring other than the consolidation of `BackoffSettings` into `RetrySettings`?

**Comment by michaelbausor** [comment]
The only other important aspect is the proposed usage by generated clients, which is documented on `RetrySettings` - we want to make sure we are happy with how users will specify retries.

File: src/RetrySettings.php [hunk 1]
```diff
@@ -102,13 +102,13 @@
...
- *     'rpc_timeout_multiplier' => 1.0,
- *     'max_rpc_timeout_millis' => 20000,
- *     'total_timeout_millis' => 600000,
+ *     'initialRetryDelayMillis' => 100,
+ *     'retryDelayMultiplier' => 1.3,
+ *     'maxRetryDelayMillis' => 60000,
+ *     'initialRpcTimeoutMillis' => 20000,
+ *     'rpcTimeoutMultiplier' => 1.0,
+ *     'maxRpcTimeoutMillis' => 20000,
+ *     'totalTimeoutMillis' => 600000,
```
Comment (bshaffer) [inline src/RetrySettings.php]: This example should include `retryableCodes`, since this is a required parameter and no example exists.

**Comment by michaelbausor** [comment]
Please see https://github.com/googleapis/gax-php/pull/102 instead

**Pull request closed without merging**